products/public-websites/cerner/postmortem/missing-content-schedule-appt-page.md

```diff
@@ -0,0 +1,103 @@
+# Postmortem for sidebar not accessible on mobile devices
+
+Date: 08/17/2020
+
+Authors: Kelson Adams
+
+Status: Resolved
+
+## What happened?
+
+In the morning of 08/17/2020, [a PR was merged](https://github.com/department-of-veterans-affairs/vets-website/pull/13882/files#diff-fbc145d9742763ba3cf52c0de9768e98R429) that updated legacy content on [www.va.gov/health-care/schedule-view-va-appointments](https://www.va.gov/health-care/schedule-view-va-appointments/) in response to [this request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/12446#issuecomment-674324572).
+
+When the PR was deployed at 3:15pm EST, Danielle Thierry then discovered it at 4:05pm EST. (2:05pm MST = 4:05pm EST)
+
+![Discovery message](./discovery.png)
+
+Traffic was almost immediately observed to go down to 0 for VAOS:
+
+![Grafana](./grafana.png)
+
+When Danielle raised the issue to Kelson Adams, Kelson opened up the dev console on the page [www.va.gov/health-care/schedule-view-va-appointments](https://www.va.gov/health-care/schedule-view-va-appointments/) and found that there was a runtime error with the common `<Telephone />` component on that page where there was a missing prop called `pattern` that is _ONLY_ required for 3-digit phone numbers (in this case, it was TTY 711 numbers). A [PR was sent up to fix this](https://github.com/department-of-veterans-affairs/vets-website/pull/13896) 5 minutes later at 4:10pm EST. The PR was approved by VSP and merged at 5pm EST.
+
+An [off-schedule deploy was requested](https://dsva.slack.com/archives/CBU0KDSB1/p1597695472119100?thread_ts=1597695207.118200&cid=CBU0KDSB1) at 4:17pm EST, which was then executed at 5:52pm EST. Following the fix being deployed at that time, traffic resumed as normal to VAOS.
+
+## Background
+
+A request was made to update the current content on the page [www.va.gov/health-care/schedule-view-va-appointments](https://www.va.gov/health-care/schedule-view-va-appointments/) with [this GitHub comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/12446#issuecomment-674324572).
+
+> we separately need to update the legacy content for this schedule and view appointments page for the Monday 8.17 deploy. The VA appointments tool is adding community care appointments that day, but we won't be quite ready to just switch to the unauthenticated Cerner content. I'm attaching the needed changes to the legacy page. Can you update these or show me how to update Monday morning? Thank you!
+
+[This request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/12446#issuecomment-674324572)
```

Comment:
I think we can zoom out here a little to describe how FE code is essentially the whole page, which I think is why this is so complex.

```suggestion
`/health-care/schedule-view-va-appointments/` serves as a gateway to the VA Online Scheduling health care application. The content on the page is housed largely in vets-website as a React widget. The reason for this is that the page content has various states based on the user's authentication status and whether the user's facilities are registered in the upcoming Cerner health care launch. This results in a fair amount of frontend logic, and means that a change to the content on that page has to be coordinated between a VA.gov writer and a frontend engineer in order to be written correctly and published at the correct time.

Via a [GitHub comment]([this GitHub comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/12446#issuecomment-674324572)), a request was made to a frontend engineer on the VSA Public Websites team to update the content on the scheduling page to promote new functionality offered by VAOS. The Public Websites team opened a pull request containing the latest content, unknowingly creating a runtime error deep in the application code. 
```